### PR TITLE
Breaking snare.py and improve project structure 

### PR DESCRIPTION
**Pull request by viskey98**
This PR will break snare.py as a part of snare reconstruction and improve the project structure. This PR is based on the previous PR  `Break server.py`. This PR can be re-based again after the previous PR gets merged (if required). :)

### CODE REVIEW

File: bin/clone [hunk 1]
```diff
@@ -27,7 +27,7 @@
 import cssutils
 import yarl
 from bs4 import BeautifulSoup
-import logger
+from snare.utils import logger
 import logging
```
Comment (afeena) [inline bin/clone]: bin file is running file, it shouldn't contain class of the cloner

File: snare/startup.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		if not os.path.exists('/opt/snare'):
+			os.mkdir('/opt/snare')
+		if not os.path.exists('/opt/snare/pages'):
+			os.mkdir('/opt/snare/pages')
+		# Write pid to pid file
+		with open('/opt/snare/snare.pid', 'wb') as pid_fh:
+			pid_fh.write(str(os.getpid()).encode('utf-8'))
+		# Config file
+		if not os.path.exists('/opt/snare/snare.cfg'):
+			create_initial_config()
```
Comment (afeena) [inline snare/startup.py]: unresolved reference

File: snare/startup.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		# Read or create the sensor id
+		uuid_file_path = '/opt/snare/snare.uuid'
+		if os.path.exists(uuid_file_path):
+			with open(uuid_file_path, 'rb') as uuid_fh:
+				snare_uuid = uuid_fh.read()
+			return snare_uuid
+		else:
+			with open(uuid_file_path, 'wb') as uuid_fh:
+				snare_uuid = str(uuid.uuid4()).encode('utf-8')
```
Comment (afeena) [inline snare/startup.py]: unresolved reference

File: snare/startup.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+			except TimeoutError:
+				print('timeout fetching the repository version')
+			else:
+				if diff_list:
+					print('you are running an outdated version, SNARE will be updated and restarted')
+					repo.git.reset('--hard')
+					repo.heads.master.checkout()
+					repo.git.clean('-xdf')
+					repo.remotes.origin.pull()
+					pip.main(['install', '-r', 'requirements.txt'])
```
Comment (afeena) [inline snare/startup.py]: unresolved reference

File: snare/startup.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+					os.execv(sys.executable, [sys.executable, __file__] + sys.argv[1:])
```
Comment (afeena) [inline snare/startup.py]: unresolved reference

File: snare/startup.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+					return
+				else:
+					print('you are running the latest version')
+				time.sleep(timeout)
```
Comment (afeena) [inline snare/startup.py]: unresolved reference

File: snare/startup.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
+import asyncio
```
Comment (afeena) [inline snare/startup.py]: there is no need to move these code into separate file, you can do it in the bin file

File: snare/tanner_handler.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+                    content = '<html><body></body></html>'
+                    content_type = r'text\html'
+
+                soup = BeautifulSoup(content, 'html.parser')
+                script_tag = soup.new_tag('div')
+                script_tag.append(BeautifulSoup(payload_content['value'], 'html.parser'))
+                soup.body.append(script_tag)
+                content = str(soup).encode()
+            else:
+                content_type = mimetypes.guess_type(payload_content['value'])[0]
```
Comment (afeena) [inline snare/tanner_handler.py]: unresolved reference

File: snare/tanner_handler.py [hunk 9]
```diff
@@ -0,0 +1,139 @@
+import asyncio
```
Comment (afeena) [inline snare/tanner_handler.py]: unused import

File: snare/utils/timeout_parser.py [hunk 10]
```diff
@@ -0,0 +1,16 @@
+def parse_timeout(timeout):
```
Comment (afeena) [inline snare/utils/timeout_parser.py]: you can combine different helper functions in one file, no need to make separate file for each function

File: snare/html_handler.py [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    def __init__(self, no_dorks, tanner):
+        self.no_dorks = no_dorks
+        self.dorks = []
+        self.logger = logging.getLogger(__name__)
+        self.tanner = tanner
+
+    async def get_dorks(self):
+        dorks = None
+        try:
+            async with aiohttp.ClientSession() as session:
```
Comment (afeena) [inline snare/html_handler.py]: unresolved reference. aiohttp import

File: snare/html_handler.py [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+                r = await session.get(
+                    'http://{0}:8090/dorks'.format(self.tanner), timeout=10.0
+                )
+                try:
+                    dorks = await r.json()
+                except json.decoder.JSONDecodeError as e:
```
Comment (afeena) [inline snare/html_handler.py]: json not imported

File: snare/server.py [hunk 13]
```diff
@@ -0,0 +1,103 @@
+import argparse
```
Comment (afeena) [inline snare/server.py]: delete unused import

File: bin/snare [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+"""
+import argparse
+import asyncio
+import configparser
+import json
+import mimetypes
```
Comment (afeena) [inline bin/snare]: delete unused import

File: bin/snare [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+        print('can\'t create meta tag')
+    else:
+        add_meta_tag(args.page_dir, meta_info[args.index_page]['hash'], config)
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(startup.check_tanner())
+
+    pool = ProcessPoolExecutor(max_workers=multiprocessing.cpu_count())
+    compare_version_fut = None
+    if args.auto_update is True:
+        timeout = startup.parse_timeout(args.update_timeout)
```
Comment (afeena) [inline bin/snare]: you moved parse_timeout into utils

File: snare/tanner_handler.py [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+class TannerHandler():
+    def __init__(self, run_args, meta, snare_uuid):
+        self.run_args = run_args
+        self.meta = meta
+        self.dir = '/opt/snare/pages/{}'.format(run_args.page_dir)
+        self.snare_uuid = snare_uuid
+        self.html_handler = HtmlHandler(run_args.no_dorks, run_args.tanner)
+        self.logger = logging.getLogger(__name__)
+
+    async def submit_slurp(self, data):
```
Comment (afeena) [inline snare/tanner_handler.py]: doesn't belong to tanner

File: snare/tanner_handler.py (hunk 9 above)
Comment (viskey98) [reply snare/tanner_handler.py]: I think we have used asyncio in all `async def` syntax. What do you think ?

File: snare/tanner_handler.py (hunk 8 above)
Comment (viskey98) [reply snare/tanner_handler.py]: fixed

File: snare/html_handler.py (hunk 11 above)
Comment (viskey98) [reply snare/html_handler.py]: fixed

File: snare/html_handler.py (hunk 12 above)
Comment (viskey98) [reply snare/html_handler.py]: fixed

**Comment by viskey98** [comment]
Now I have done all the suggested changes.
Note: 
* Now I have let cloner as clone.py in snare folder itself. This is because the cloner will be re-written in the next pr and then added to the setup structure.
* I have combined all the snare helper functions (including version_manager, converter etc) into utils/snare_helper.py
I have tried to make all changes in previous prs and rebased them consecutively, if any changes in this pr are by chance missing in previous prs, they can be over-written when we merge this pr, so I don't think it will be a problem. :)

File: snare/utils/snare_helper.py [hunk 17]
```diff
@@ -0,0 +1,85 @@
+from distutils.version import StrictVersion
+from bs4 import BeautifulSoup
+import os
+
+
+class SnareHelper:
```
Comment (afeena) [inline snare/utils/snare_helper.py]: this class contains a lot of different functions which is not connected. Why you need a class?

File: snare/utils/snare_helper.py [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+        timeouts_coeff = {
+            'M': 60,
+            'H': 3600,
+            'D': 86400
+        }
+
+        form = self.timeout[-1]
+        if form not in timeouts_coeff.keys():
+            print('Bad timeout format, default will be used')
+            parse_timeout('24H')
```
Comment (afeena) [inline snare/utils/snare_helper.py]: now this function is a class function (which is bad approach in my opinion) and you need to use self

File: snare/utils/snare_helper.py [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+    def convert(self):
+        files_to_convert = []
+        for (dirpath, dirnames, filenames) in walk(self.path):
+            for fn in filenames:
+                files_to_convert.append(os.path.join(dirpath, fn))
+
+        for fn in files_to_convert:
+            path_len = len(self.path)
+            file_name = fn[path_len:]
+            m = hashlib.md5()
```
Comment (afeena) [inline snare/utils/snare_helper.py]: hashlib  - unresolved reference

File: snare/utils/snare_helper.py [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+            m.update(fn.encode('utf-8'))
+            hash_name = m.hexdigest()
+            self.meta[file_name] = {'hash': hash_name, 'content_type': mimetypes.guess_type(file_name)[0]}
+            shutil.copyfile(fn, os.path.join(self.path, hash_name))
```
Comment (afeena) [inline snare/utils/snare_helper.py]: shutil - unresolved reference

File: snare/utils/snare_helper.py [hunk 21]
```diff
@@ -0,0 +1,85 @@
...
+    def check_compatibility(self, tanner_version):
+        min_version = self.version_mapper[self.version][0]
+        max_version = self.version_mapper[self.version][1]
+        if not (StrictVersion(min_version) <= StrictVersion(tanner_version) <= StrictVersion(max_version)):
+            raise RuntimeError("Wrong tanner version: {}. Compatible versions are {} - {}"
+                               .format(tanner_version, min_version, max_version))
+
+    def convert(self):
+        files_to_convert = []
+        for (dirpath, dirnames, filenames) in walk(self.path):
```
Comment (afeena) [inline snare/utils/snare_helper.py]: walk - unresolved reference

File: snare/utils/snare_helper.py [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+            file_name = fn[path_len:]
+            m = hashlib.md5()
+            m.update(fn.encode('utf-8'))
+            hash_name = m.hexdigest()
+            self.meta[file_name] = {'hash': hash_name, 'content_type': mimetypes.guess_type(file_name)[0]}
+            shutil.copyfile(fn, os.path.join(self.path, hash_name))
+            os.remove(fn)
+
+        with open(os.path.join(self.path, 'meta.json'), 'w') as mj:
+            json.dump(self.meta, mj)
```
Comment (afeena) [inline snare/utils/snare_helper.py]: json - unresolved reference

File: snare/server.py [hunk 23]
```diff
@@ -0,0 +1,106 @@
+import asyncio
```
Comment (afeena) [inline snare/server.py]: unused imports

**Comment by afeena** [comment]
Please, check your imports carefully

File: snare/server.py (hunk 23 above)
Comment (viskey98) [reply snare/server.py]: I think we need asyncio for all `async def` syntax, what do you think

**Comment by viskey98** [comment]
@afeena I have revised and made the suggested changes please look into it :)

**Comment by viskey98** [comment]
Fixed by latest PR #141

**Pull request closed without merging**